lib/services/wallet_connect_service.dart

```diff
@@ -0,0 +1,641 @@
+import 'package:flutter/material.dart';
+import 'package:logging/logging.dart';
+import 'package:walletconnect_flutter_v2/walletconnect_flutter_v2.dart';
+import 'package:window_manager/window_manager.dart';
+import 'package:zenon_syrius_wallet_flutter/blocs/blocs.dart';
+import 'package:zenon_syrius_wallet_flutter/blocs/wallet_connect/wallet_connect_pairings_bloc.dart';
+import 'package:zenon_syrius_wallet_flutter/blocs/wallet_connect/wallet_connect_sessions_bloc.dart';
+import 'package:zenon_syrius_wallet_flutter/main.dart';
+import 'package:zenon_syrius_wallet_flutter/model/model.dart';
+import 'package:zenon_syrius_wallet_flutter/utils/functions.dart';
+import 'package:zenon_syrius_wallet_flutter/utils/utils.dart';
+import 'package:zenon_syrius_wallet_flutter/widgets/main_app_container.dart';
+import 'package:zenon_syrius_wallet_flutter/widgets/reusable_widgets/dialogs.dart';
+import 'package:zenon_syrius_wallet_flutter/widgets/reusable_widgets/icons/link_icon.dart';
+import 'package:znn_sdk_dart/znn_sdk_dart.dart';
+
+class WalletConnectService {
+  WalletConnectService._internal();
+
+  factory WalletConnectService() => _instance;
+
+  static final WalletConnectService _instance =
+      WalletConnectService._internal();
+
+  final List<int> _idSessionsApproved = [];
+
+  late Web3Wallet _wcClient;
+  late BuildContext _context;
+
+  final List<SessionData> dAppsActiveSessions = [];
+
+  set context(BuildContext context) => _context = context;
+
+  final _walletLockedError = const WalletConnectError(
+    code: 9000,
+    message: 'Wallet is locked',
+  );
+
+  Future<void> initClient() async {
+    if (kWcProjectId.isNotEmpty) {
+      _wcClient = await Web3Wallet.createInstance(
+        projectId: kWcProjectId,
+        metadata: const PairingMetadata(
+          name: 's y r i u s',
+          description: 'A wallet for interacting with Zenon Network',
+          url: 'https://zenon.network',
+          // TODO: add Zenon icon
+          icons: ['https://avatars.githubusercontent.com/u/37784886'],
```

Comment:
I don't remember seeing this image anywhere, but maybe we should replace it with a branded one anyways?